src/ext/NetFx/ca/netfxca.cpp

```diff
@@ -879,3 +892,124 @@ extern "C" UINT __stdcall ExecNetFx(
     return WcaFinalize(er);
 }
 
+/******************************************************************
+ DotNetCompatibilityCheck - entry point for NetFx Custom Action
+
+*******************************************************************/
+extern "C" UINT __stdcall DotNetCompatibilityCheck(
+    __in MSIHANDLE hInstall
+    )
+{
+//    AssertSz(FALSE, "debug DotNetCompatibilityCheck");
+
+    HRESULT hr = S_OK;
+    UINT er = ERROR_SUCCESS;
+
+    PMSIHANDLE hView = NULL;
+    PMSIHANDLE hRec = NULL;
+    LPWSTR pwzPlatform = NULL;
+    LPWSTR pwzNetCoreCheckBinaryId = NULL;
+    LPWSTR pwzNetCoreCheckFilePath = NULL;
+    LPWSTR pwzRuntimeType = NULL;
+    LPWSTR pwzVersion = NULL;
+    LPWSTR pwzRollForward = NULL;
+    LPWSTR pwzProperty = NULL;
+    LPWSTR pwzCommandLine = NULL;
+    HANDLE hProcess = NULL;
+    DWORD dwExitCode = 0;
+
+    // Initialize
+    hr = WcaInitialize(hInstall, "DotNetCompatibilityCheck");
+    ExitOnFailure(hr, "failed to initialize");
+
+    // If Wix4NetFxDotNetCheck table doesn't exist skip the rest of this custom action
+    hr = WcaTableExists(L"Wix4NetFxDotNetCheck");
+    if (S_FALSE == hr)
+    {
+        hr = S_OK;
+        ExitFunction();
+    }
+    ExitOnFailure(hr, "failed to check if table Wix4NetFxDotNetCheck exists");
+
+    // Open view on .NET compatibility check table
+    hr = WcaOpenExecuteView(vscDotNetCompatibilityCheckQuery, &hView);
+    ExitOnFailure(hr, "failed to open view on Wix4NetFxDotNetCheck table");
+
+    // Go through all records and run NetCorCheck.exe for each
+    while (S_OK == (hr = WcaFetchRecord(hView, &hRec)))
+    {
+        // Extract NetCoreCheck.exe for platform to temp directory
+        hr = WcaGetRecordString(hRec, platform, &pwzPlatform);
+        ExitOnFailure(hr, "failed to get Wix4NetFxDotNetCheck.Platform");
+
+        hr = StrAllocFormatted(&pwzNetCoreCheckBinaryId, L"Wix4NetCheck_%ls", pwzPlatform);
+        ExitOnFailure(hr, "failed to get NetCoreCheck binary id for platform %ls", pwzPlatform);
+
+        hr = PathCreateTempFile(NULL, NULL, 0, L"", 0, &pwzNetCoreCheckFilePath, NULL);
```

Comment:
Two things here:

1. We need to extract the binary to a randomly named directory to prevent the exe from being DLL hijacked. Since we don't control the exe this simple layer of protection seems prudent.

2.  Doesn't the file name need to be extracted with `.exe` extension so it can be executed later?

Reply:
I've made the requested changes.

1. This directory has a randomly generated name from range NCC_0000 - NCC_7FFF.
2. Surprisingly, `.tmp` was working fine. Since there is now a randomly generated directory name, I went back to naming the extracted file `NetCoreCheck.exe`

Reply:
1. That isn't going to be random enough. The issue is that an attacker can seed all those directories in the System %TMP% folder with an attack DLL then wait for an elevated `NetCoreCheck.exe` to be run. I still recommend doing what Burn does and use a GUID (`GuidFixedCreate()` exists for that reason in `guidutil.h`). A GUID is _entirely_ non-guessable.

2. I am surprised `::CreateProcess()` cares less than I thought it would. 😄

Reply:
> 1. That isn't going to be random enough. The issue is that an attacker can seed all those directories in the System %TMP% folder with an attack DLL then wait for an elevated `NetCoreCheck.exe` to be run. I still recommend doing what Burn does and use a GUID (`GuidFixedCreate()` exists for that reason in `guidutil.h`). A GUID is _entirely_ non-guessable.

I agree, GUID was the first thing that came to my mind and functions provided by `guidutil.h` were the first thing I tried. However using either `GuidCreate` or `GuidFixedCreate` was resulting in LNK2019 error `error LNK2019: unresolved external symbol`

![image](https://user-images.githubusercontent.com/9998547/197648037-3f023d98-0761-4b75-9ea6-f475d05a4135.png)

Reply:
Ahh, I think you need to add `rpc.lib` (I'm away from my computer so that name isn't exactly it, but close, you will find it is used elsewhere) to the additional libraries in the vcxproj.

Reply:
Thanks a lot! Adding `rpcrt4.lib` worked. New changes pushed.